File: Tree Data Structure/FBinary Tree.cpp
//to check the given tree is binary or not
#include <bits/stdc++.h>
using namespace std;

struct Node{
	int data;
	Node *right,*left;
};
struct Node *node(char k)
{
	struct Node *node=(struct Node*)malloc(sizeof(struct Node));
	node->data=k;
	node->right=node->left=NULL;
	return node;
}
bool isfull(struct Node *root)
{
	if(root==NULL)
		return true;
	if(root->left==root->right==NULL)
		return true;
	if(root->left!=NULL && root->right!=NULL)
		return (isfull(root->left)&&isfull(root->right));
	return false;

}
int main() {
  struct Node *root = NULL;
  root = node(1);
  root->left = node(2);
  root->right = node(3);
  root->left->left = node(4);
  root->left->right = node(5);
  root->left->right->left = node(6);
  root->left->right->right = node(7);

  if (isfull(root))
    cout << "The tree is a full binary tree\n";
  else
    cout << "The tree is not a full binary tree\n";
return 0;
}

File: Tree Data Structure/FBinary Tree.py
#to check the given tree is a binary tree.
class Node(object):
	"""docstring for Node"""
	def __init__(self, arg):
		self.left=None
		self.right=None
		self.item=arg

def isfull(root):
	if root is None:
		return True
	if root.left is None and root.right is None:
		return True
	if root.left is not None and root.right is not None:
		return (isfull(root.left)and isfull(root.right))

	return False

root = Node(1)
root.rightChild = Node(3)
root.leftChild = Node(2)

root.leftChild.leftChild = Node(4)
root.leftChild.rightChild = Node(5)
root.leftChild.rightChild.leftChild = Node(6)
root.leftChild.rightChild.rightChild = Node(7)

if isfull(root):
    print("The tree is a full binary tree")
else:
    print("The tree is not a full binary tree")